src/web/controllers/index.js

```diff
@@ -405,31 +420,46 @@ router.get('/rankings', function (req, res) {//"Please login first"
         if (req.session) {
             if (req.session.account_id) {
                 var acc_id = req.session.account_id;
+
+                // Debugging: Log account ID
+                console.log("Account ID:", acc_id);
+
                 req.db.connection.getConnection().then(conn => {
                     conn.query('SELECT g.Id FROM users u INNER JOIN guild_member m ON m.UserId = u.IdAcc LEFT JOIN guild g ON g.Id = m.Id WHERE u.IdAcc = ?', [acc_id])
                         .then(rowss => {
-                        conn.release();
-                        if (rowss[0].length > 0) {
-                            let res00 = rowss[0];
-                            conn.query('SELECT u.game_id, u.IdAcc, u.rank, u.gp from guild_member m INNER JOIN guild g ON m.Id = g.Id INNER JOIN users u ON m.UserId = u.IdAcc WHERE g.Id = ? ORDER BY u.gp DESC', [res00[0].Id])
-                                .then(rows => {
-                                conn.release();
-                                if (rows[0].length > 0) {
-                                    let res0 = rows[0];
-                                    data.push(6);
-                                    data.push(1);
-                                    for (var u in res0) {
-                                        data.push(/*u, */res0[u].gp, res0[u].rank, res0[u].game_id);
-                                    }
-                                    res.send(JSON.stringify(data));
-                                } else {
-                                    res.send(JSON.stringify("You are not in a guild"));
-                                }
-                            });
-                        } else {
-                            res.send(JSON.stringify("You are not in a guild"));
-                        }
-                    });
+                            conn.release();
+                            if (rowss[0].length > 0) {
+                                let res00 = rowss[0];
+                                conn.query('SELECT u.game_id, u.IdAcc, u.rank, u.gp from guild_member m INNER JOIN guild g ON m.Id = g.Id INNER JOIN users u ON m.UserId = u.IdAcc WHERE g.Id = ? ORDER BY u.gp DESC', [res00[0].Id])
+                                    .then(rows => {
+                                        conn.release();
```

Comment:
handling nested structures only generates callbackhells and is complicated to handle. Please improve this